Software/spartan/src/sensors/sensor.h

```diff
@@ -29,6 +31,7 @@ namespace spartan {
         // Data getters
         int getBusID() const;
         int getInstance() const;
+		const long frequency = 1000; 
```

Comment:
Is there a particular reason why we have our "default" value as `1000` milliseconds?

When making some of the other changes I mentioned in this review, remember that we want our `Sensor`-derived classes to themselves define their own frequencies... using an [initializer list](https://en.cppreference.com/w/cpp/language/constructor) for this would probably be the best option.